examples/cluster/plot_kmeans_assumptions.py

```diff
@@ -4,66 +4,116 @@
 ====================================
 
 This example is meant to illustrate situations where k-means will produce
-unintuitive and possibly unexpected clusters. In the first three plots, the
-input data does not conform to some implicit assumption that k-means makes and
-undesirable clusters are produced as a result. In the last plot, k-means
-returns intuitive clusters despite unevenly sized blobs.
+unintuitive and possibly undesirable clusters.
+
+- Incorrect number of blobs: in a real setting there is no uniquely defined
+  **true** number of clusters. An appropriate number of clusters has to be
+  decided from data-based criteria and knowledge of aim.
+- Anisotropically distributed blobs: k-means consists of minimizing sample's
+  euclidean distances to the centroid of the cluster they are assigned
+  to. As a consequence, k-means is more appropriated for clusters that are
+  normally distributed with a spherical covariance matrix.
```

Comment:
Does the average person on the street know what a "spherical covariance matrix" is? I don't :( Is there a way to say the same thing with simpler words, without it becoming super long?

---

I imagine a spherical covariance matrix is one you print out and then crumple up the paper into a ball, hence spherical ;)

Reply:
I am still hesitating about the simplest yet clearest wording. Which one do you like the best?

a) "k-means is more appropriate for clusters that are isotropic and normally distributed."

b) "k-means is more appropriate for clusters that are normally distributed and have spherical contours (as opposed to Gaussian distributions with ellipsoidal contours)."

Reply:
Probably (b) because it includes a counter example. Helps me understand what is meant. Maybe it makes sense to include a link/reference to the section of the docs about rescaling features before feeding them to `KMeans`. Something like "and this is why it is smart to scale your features."?

Reply:
By addressing [this comment](https://github.com/scikit-learn/scikit-learn/pull/24970/files#r1027279467) by @glemaitre I changed the narrative of the example. Now I introduce the concept of spherical/elliptical gaussians in the first section, before presenting the scenarios where the concept is required, what do you think?

About rescaling features it does not seem to have an effect in this particular case, as both features have the same range. In this sense I rather added a comment on another possible issue: sparse high-dimensional spaces. If we really want to pass a message on the importance of scaling, maybe we can try reviving the discussion on #12282.

Reply:
Works for me 👍